test/python/test_replica.py

```diff
@@ -0,0 +1,317 @@
+
+import pytest
```

Comment:
The files do not seem to be properly formatted. I've used `black` on the other files and its also run by the precommit hooks

Reply:
reformatted via black as per precommit hook